src/java.base/share/classes/java/lang/StringTemplate.java

```diff
@@ -0,0 +1,615 @@
+     * String s2 = "xyz";
+     * String sc = s1 + s2;
+     * assert Objects.equals(sc, "abcxyz");
+     * }
+     * the last character {@code "c"} from the first string is juxtaposed with the first
+     * character {@code "x"} of the second string. The same would be true of combining
+     * {@link StringTemplate StringTemplates}.
+     * {@snippet lang ="java":
+     * StringTemplate st1 = RAW."a\{}b\{}c";
+     * StringTemplate st2 = RAW."x\{}y\{}z";
+     * StringTemplate st3 = RAW."a\{}b\{}cx\{}y\{}z";
+     * StringTemplate stc = StringTemplate.combine(List.of(st1, st2));
+     *
+     * assert Objects.equals(st1.fragments(), List.of("a", "b", "c"));
+     * assert Objects.equals(st2.fragments(), List.of("x", "y", "z"));
+     * assert Objects.equals(st3.fragments(), List.of("a", "b", "cx", "y", "z"));
+     * assert Objects.equals(stc.fragments(), List.of("a", "b", "cx", "y", "z"));
+     * }
+     * Values lists are simply concatenated to produce a single values list.
+     * The result is a well-formed {@link StringTemplate} with n+1 fragments and n values, where
+     * n is the total of number of values across all the supplied
+     * {@link StringTemplate StringTemplates}.
+     *
+     * @param stringTemplates  list of {@link StringTemplate}
+     *
+     * @return combined {@link StringTemplate}
+     *
+     * @throws NullPointerException if stringTemplates is null or if any of the
+     * its elements are null
+     *
+     * @implNote If {@code stringTemplates.size() == 0} then a {@link StringTemplate} with
+     * an empty fragment and no values is returned, as if invoking
+     * <code>StringTemplate.of("")</code> . If {@code stringTemplates.size() == 1}
+     * then the first element of the list is returned unchanged.
+     */
+    static StringTemplate combine(List<StringTemplate> stringTemplates) {
+        JavaTemplateAccess JTA = SharedSecrets.getJavaTemplateAccess();
+        return JTA.combine(stringTemplates.toArray(new StringTemplate[0]));
+    }
+
+    /**
+     * This {@link Processor} instance is conventionally used for the string interpolation
+     * of a supplied {@link StringTemplate}.
+     * <p>
+     * For better visibility and when practical, it is recommended that users use the
+     * {@link StringTemplate#STR} processor instead of invoking the
+     * {@link StringTemplate#interpolate()} method.
+     * Example: {@snippet :
+     * int x = 10;
+     * int y = 20;
+     * String result = STR."\{x} + \{y} = \{x + y}"; // @highlight substring="STR"
+     * }
+     * In the above example, the value of {@code result} will be {@code "10 + 20 = 30"}. This is
+     * produced by the interleaving concatenation of fragments and values from the supplied
+     * {@link StringTemplate}. To accommodate concatenation, values are converted to strings
+     * as if invoking {@link String#valueOf(Object)}.
+     * @implNote {@link StringTemplate#STR} is statically imported implicitly into every
```

Comment:
IMO the remark regarding  STR being auto-imported works better as an apiNote rather than an implNote.

Regarding interning, I think an implSpec of "may or may not be interned" provides maximum flexibility to the implementation.

Reply:
Changed